### PR TITLE
feat(experimental-utils): extract `isNodeOfTypes` out of `ast-utils`' `predicates`

### PR DESCRIPTION
**Pull request by MichaelDeBoey**

### CODE REVIEW

File: packages/experimental-utils/src/ast-utils/predicates.ts [hunk 1]
```diff
@@ -6,6 +6,12 @@ const isNodeOfType =
     node: TSESTree.Node | null | undefined,
   ): node is TSESTree.Node & { type: NodeType } =>
     node?.type === nodeType;
+const isNodeOfTypes =
+  <NodeTypes extends AST_NODE_TYPES[]>(nodeTypes: NodeTypes) =>
```
Comment (bradzacher) [inline packages/experimental-utils/src/ast-utils/predicates.ts:10]:
```suggestion
  <NodeTypes extends readonly AST_NODE_TYPES[]>(nodeTypes: NodeTypes) =>
```

File: packages/experimental-utils/src/ast-utils/predicates.ts [hunk 2]
```diff
@@ -78,89 +84,48 @@ function isTypeAssertion(
...
-  | TSESTree.TSEmptyBodyFunctionExpression
-  | TSESTree.TSFunctionType
-  | TSESTree.TSMethodSignature {
-  return isFunction(node) || isFunctionType(node);
-}
+const functionTypes = [
+  AST_NODE_TYPES.ArrowFunctionExpression,
+  AST_NODE_TYPES.FunctionDeclaration,
+  AST_NODE_TYPES.FunctionExpression,
+];
```
Comment (bradzacher) [inline packages/experimental-utils/src/ast-utils/predicates.ts:91]:
```suggestion
] as const;
```

**Review by bradzacher — COMMENTED** [review]
you'll need to `as const` the arrays to ensure that TS retains the individual enum members listed in the arrays for typing (or else the arrays will be inferred as `AST_NODE_TYPES[]`)

**Review by bradzacher — DISMISSED** [review]
we should probably make all of these `Set`s for guaranteed `O(1)` lookups but they're small enough that it probably doesn't matter.

**Comment by bradzacher** [comment]
@MichaelDeBoey - I just flicked this to a non-draft - just triple checking before I merge it - it's ready, yeah? Nothing else you wanted?

**Comment by MichaelDeBoey** [comment]
@bradzacher this one can be merged for me

**Pull request merged**